linera-core/src/client/mod.rs

```diff
@@ -1033,15 +1033,20 @@ where
         self.state_mut()
             .set_pending_block(proposal.content.block.clone());
         let required_blob_ids: HashSet<BlobId> = value.inner().required_blob_ids();
-        let proposed_blobs = proposal.blobs.clone();
+        let required_blobs = proposal
```

Comment:
Maybe we should rename the `fn required_blob_ids` to something including _oracle_ in the name? b/c _required_ is quite ambiguous and could also mean "blob ids used by this transaction" (that would include published and read blobs).

Reply:
Maybe `oracle_blob_ids`?

I also realized that _publishing_ blobs should not count as an oracle call.

Reply:
Sounds good.

I didn't even realize it does ;) but I agree it shouldn't.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/2972/commits/53b6807582774d3a4140f0dbf70f2d8b4a77140d.
I kept the name `required_blob_ids` in most places and changed the implementation to include the published blobs: That's what we need in all call sites.

Reply:
And updated a test in https://github.com/linera-io/linera-protocol/pull/2972/commits/647676bdd79a4f6f2501da555a372c1f30878f97.

(This test already verifies that when we create the application on _another_ chain, that _is_ still an oracle call, as expected.)